src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/ArtifactsSizeTest.cs

```diff
@@ -0,0 +1,155 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Collections;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using System.Formats.Tar;
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.DotNet.SourceBuild.SmokeTests;
+
+[Trait("Category", "SdkContent")]
+public class ArtifactsSizeTest : SmokeTests
+{
+    private static readonly string BaselineFilePath = BaselineHelper.GetBaselineFilePath($"ArtifactsSizes/{Config.TargetRid}.txt");
+    private readonly Dictionary<string, long> BaselineFileContent = new();
+    private Dictionary<string, int> FilePathCountMap = new();
+    private const int SizeThresholdPercentage = 25;
+
+
+    public ArtifactsSizeTest(ITestOutputHelper outputHelper) : base(outputHelper)
+    {
+        if (File.Exists(BaselineFilePath))
+        {
+            string[] baselineFileContent = File.ReadAllLines(BaselineFilePath);
+            foreach (string entry in baselineFileContent)
+            {
+                string[] splitEntry = entry.Split(':', StringSplitOptions.TrimEntries);
+                BaselineFileContent[splitEntry[0]] = long.Parse(splitEntry[1]);
+            }
+        }
+        else
+        {
+            Assert.False(true, $"Baseline file `{BaselineFilePath}' does not exist. Please create the baseline file then rerun the test.");
+        }
+    }
+
+    [SkippableFact(new[] { Config.SourceBuiltArtifactsPathEnv, Config.SdkTarballPathEnv, Config.TargetRidEnv, Config.ExcludeArtifactsSizeEnv }, skipOnNullOrWhiteSpaceEnv: true, skipOnTrueEnv: true)]
+    public void CompareArtifactsToBaseline()
+    {
+        Assert.NotNull(Config.SourceBuiltArtifactsPath);
```

Comment:
These seem redundant given the SkippableFactAttribute.

Reply:
I agree that this is redundant; however, SdkContentTests use Assert.NotNull() with the SkippableFactAttribute:

```
[SkippableFact(new[] { Config.MsftSdkTarballPathEnv, Config.SdkTarballPathEnv }, skipOnNullOrWhiteSpaceEnv: true)]
    public void CompareMsftToSbAssemblyVersions()
    {

        Assert.NotNull(Config.MsftSdkTarballPath);
        Assert.NotNull(Config.SdkTarballPath);
      ...

```

Is it prefered to include Assert.NotNull() or not include it based on the SkippableFact & skipOnNullOrWhiteSpaceEnv combination?

Reply:
The reason it's done in the `CompareMsftToSbAssemblyVersions` test is because `Config.MsftSdkTarballPath` and `Config.SdkTarballPath` are nullable string properties. By including the `Assert` calls, it provides a hint to the compiler so that it knows, after those calls have been made, that the property values cannot be null. So later code can dereference those properties without any nullable warnings. This doesn't apply to `Config.SourceBuiltArtifactsPath` because it's already non-nullable.

Reply:
Addressed [here](https://github.com/dotnet/installer/pull/17233/commits/8ec128e7e0e58aa70e87b705d20616b29250bb04#diff-ce0a9d7a18f02288a0510a8ec86e3a5f387ad96613e0296f434de93f02c18da7R47).